codc/codc.py

```diff
@@ -87,6 +86,34 @@ def type_name(self, typ: 'DataType'):
         self.__type_names[typ] = name
         return name
 
+    def __find_library(self, lib_file_base):
```

Comment:
I really like this function, in principle. But while it extends the functionality in some ways, it breaks other things.

In particular, dlopen() when supplied with just the library name performs library lookups in all the locations an OS would do so - including those pointed at by LD_LIBRARY_PATH. This is important for integrating with, say, the module system.

sys.path does not contain all of the relevant paths.

I might suggest instead creating something along the lines of:

```
def __dlopen(self):
    
    # Find appropriate file suffix based on what platform we
    # are running on
    if 'linux' in sys.platform:
        lib_sfx = '.so'
    elif 'darwin' in sys.platform:
        lib_sfx = '.dylib'
    elif 'win' in sys.platform:
        lib_sfx = '.DLL'
    else:
        raise Exception("Don't know lib extension for platform " + sys.platform)

    lib_file = 'libodccore' + lib_sfx

    try:
        return ffi.dlopen(libname)
    except <some exception>:
        pass

    # We have been unable to locate the library using dlopen, so do
    # our own search in sys.paths

    lib_full_path = None;
    for dirpath in sys.path:
        test_path = os.path.join(dirpath, lib_file)
        if (os.path.exists(test_path)):
            lib_full_path = test_path
            break

    # lib_full_path will either be None (couldn't find the library)
    # or a full path to the library file.
    if (lib_full_path is None):
        raise Exception("Cannot find libaray file: " + lib_file)

    return ffi.dlopen(lib_full_path)
```

(n.b. not checked for correctness ... at all)

Reply:
Thanks for the feedback here. I'll look into getting this restructured to look through both sys.path and LD_LIBRARY_PATH.